Allow saving of relations with select

cfothergill opened the pull request:
I was creating a form the other day that assigned many of one model to another during creation. At the moment the this is not possible. 

In this pull request I have added an option to the DefaultForm class which switches it to use `push` instead of `save`.

noxify commented:
Hi,

thanks for your PR.

Can you add a short example for this?
I need it for the documentation.

Thanks - After this, I can merge it.

cfothergill commented:
Hi. Yes, I will do an example first thing in the morning.

I have also got a couple of modifications to Select which works with this and allows it to save to many relations. I will add this to the Pull Request too in the morning. 

For the example do you want a working project or just a few files showing how the changes work?

Thanks

cfothergill commented:
Sorry, I pressed the close button instead of the comment one.

noxify commented:
For me, some lines are enough.

noxify commented:
And.. Thanks!

cfothergill commented:
I have added the modifications to Select to the Pull Request. Basically I had a Listing Model and a Tag Model. During creation of the Listing I wanted to be able to assign Tags. As the library stands it only supported setting simple attributes and obviously during creation the ID of the Listing is unknown so I cannot associate Tags. 

I misunderstood what Eloquent's push method did so I have took that out. 

To accomplish the task I have added a `saved` method to `FormItemInterface` which is called after the forms instance is saved. This gives any Form Items the chance to save relationships that needed the instances ID. 

The updated Select Class supports all the various To Many Relations expect HasManyThrough as I didn't think that made sense. All you have to do to enable the behaviour is set `relation` to `true`. 

```php
[
    // ...
    FormItem::select('tags', 'Tags')->multi(true)->model(Tag::class)->relation(true),
    // ...
]
```

The only other modification I had to make was to `Columns` so that the `saved` method was passed through. 

Thanks
